tests/integration/testdata/seccomp_syscall_test1.c

```diff
@@ -0,0 +1,68 @@
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <sched.h>
+#include <sys/syscall.h>
+
+// We need to have a "raw" version because the glibc wrapper hides -ENOSYS from
+// us, which leads to confusing results...
+int raw_renameat2(int olddirfd, const char *oldpath, int newdirfd, const char *newpath, unsigned int flags)
+{
+	return syscall(SYS_renameat2, olddirfd, oldpath, newdirfd, newpath, flags);
```

Comment:
This will probably fail to compile on e.g. centos7.

Reply:
Indeed it fails:

```
not ok 70 runc run [seccomp -ENOSYS handling]
# (in test file tests/integration/seccomp.bats, line 32)
#   `[ "$status" -eq 0 ]' failed with status 218
# runc spec (status=0):
# 
# runc run test_busybox (status=218):
```

but the CI is green because of an issue fixed by https://github.com/opencontainers/runc/pull/2745 (cc @AkihiroSuda @mrunalp)

Reply:
This isn't failing to compile -- this is failing during execution but without a loud warning. I mentioned this [yesterday](https://github.com/opencontainers/runc/pull/2737#issuecomment-760999681) but I'm still confused why there's no errors, I'll investigate this on Monday...

Reply:
I finally figured it out -- it's because I'm missing `return 0;`. The last register value is used instead (`-38` for `-ENOSYS`) which then gets wrapped around to `218`. I guess newer GCCs add an implicit `return 0;` but not the one in CentOS 7.